web/ui/src/common/jellyTable.js

```diff
@@ -123,65 +125,89 @@
                         }
                     };
 
-                    getData = function($defer, params) {
-                        var unorderedData = data(),
-                            orderedData;
+                    getData = function ($defer, params) {
+                        var allItems = data(),
+                            totalItemCount = 0,
+                            sortedItems,
+                            tableEntries;
 
-                        // if unorderedData is an object, convert to array
-                        // NOTE: angular.isObject does not consider null to be an object
-                        if(!angular.isArray(unorderedData) && angular.isObject(unorderedData)){
-                            unorderedData = utils.mapToArr(unorderedData);
+                        if (allItems === undefined) {
 
-                        // if it's null, create empty array
-                        } else if(unorderedData === null){
-                            unorderedData = [];
-                        }
+                            // if just intitalized, show loading and default to empty array
+                            allItems = [];
+                            toggleNoData(true);
+                            return;
 
-                        // call overriden getData
-                        if(config().getData){
-                            orderedData = config().getData(unorderedData, params);
+                        } else if (angular.isObject(allItems) && !angular.isArray(allItems)) {
 
-                        // use default getData
-                        } else {
-                            orderedData = params.sorting() ?
-                                orderBy(unorderedData, params.orderBy()) :
-                                unorderedData;
-                        }
-
-                        // if no data, show loading and default
-                        // to empty array
-                        if(angular.isUndefined(orderedData)){
-                            $scope[tableID].loading = true;
+                            // single-entry tables that are not arrays pass through once (eg config files)
+                            tableEntries = utils.mapToArr(allItems);
```

Comment:
Chatted with carl via slack about some changes in `getData` that are still being looked at.